packages/flutter/lib/src/foundation/isolates.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:isolate';
+
+import 'package:meta/meta.dart';
+
+/// Signature for the callback passed to [compute].
+///
+/// {@macro flutter.foundation.compute.types}
+///
+/// Instances of [ComputeCallback] must be top-level functions, not closures or
+/// instance or static methods of classes.
+///
+/// {@macro flutter.foundation.compute.limitations}
+typedef R ComputeCallback<Q, R>(Q message);
+
+/// Spawn an isolate, run `callback` on that isolate, passing it `message`, and
+/// (eventually) return the value returned by `callback`.
+///
+/// This is useful for operations that take longer than a few milliseconds, and
+/// which would therefore risk skipping frames. For tasks that will only take a
+/// few milliseconds, consider [scheduleTask] instead.
+///
+/// {@template flutter.foundation.compute.types}
+/// `Q` is the type of the message that kicks off the computation.
+///
+/// `R` is the type of the value returned.
+/// {@endtemplate}
+///
+/// The `callback` argument must be a top-level function, not a closure or an
+/// instance or static method of a class.
+///
+/// {@template flutter.foundation.compute.limitations}
+/// There are limitations on the values that can be sent and received to and
+/// from isolates. These limitations constrain the values of `Q` and `R` that
+/// are possible. See the discussion at [SendPort.send].
+/// {@endtemplate}
+Future<R> compute<Q, R>(ComputeCallback<Q, R> callback, Q message) async {
```

Comment:
I'm wondering if returning `Stream<R>` would be more conducive to designing more effective solutions to jank. If each element of the stream is sent as a separate message, it would be processed in its own event cycle thus not blocking the UI (e.g. it wouldn't jank an animating progress indicator). For example, we could have `_parseLicenses` return `Stream<LicenseEntry>` and deserialize one entry per event instead of parsing the whole list of entries in one event. Another benefit is that `Stream` can be cancelled, causing the isolate to shut down (e.g. to implement a timeout). We shouldn't lose in generality as `Future<R>` can be emulated using `Stream<R>` (e.g. `stream.first`).

Reply:
Yeah I tried adding Stream support just before submitting this (and going to bed), but in the few minutes I had before bed time, I couldn't get it to really work well. I can push at it some more.

Reply:
Actually, we can add support for Stream and Future later, since they are not supported right now. We could dynamically check the type of the returned value, and if we find that it's a Stream or a Future, we send them down specialized code paths.

Reply:
ok, sounds good